status/discover.go

```diff
@@ -0,0 +1,62 @@
+package status
+
+import (
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"io/ioutil"
+)
+
+// The Kiali ServiceAccount token.
+var saToken string
+
+var clientFactory kubernetes.ClientFactory
+
+func DiscoverJaeger() (url string, err error) {
+	if config.Get().ExternalServices.Jaeger.URL != "" {
+		return config.Get().ExternalServices.Jaeger.URL, nil
+	}
+	return discoverUrlService(config.Get().ExternalServices.Jaeger.ServiceNamespace, config.Get().ExternalServices.Jaeger.Service)
+}
+
+func DiscoverGrafana() (url string, err error) {
```

Comment:
Note for Grafana, it links can be disabled with config.DisplayLink.
So if !DisplayLink it should return without trying to get URL I think.

Reply:
Great !!! Thanks